std/machines/small_field/pointer_arith.asm

```diff
@@ -0,0 +1,38 @@
+/// Increments a pointer by one full 32-bit word.
```

Comment:
Both this comment and the name of the function: it returns a constraint that forces a difference of one, it does not really increment it. This might be minor and it might also be more complicated to describe, but I was really wondering if this is a prover function that computes a witness, or if it adds/creates a constraint.

Reply:
On one hand, I agree this is bi-directional set of constraints, on the other hand, I doubt I can use this function to decrement instead of increment, as it would be too hard for witgen to solve.

Do you have a suggestion for a better name?

Reply:
I fixed the comment.

Reply:
I ran the Gröbner Basis, and going on the opposite direction from `post_low = 0`, I get that I can either get `pre_low = 0xfffc` with overflow (the correct answer) or `pre_low = -4` with no overflow (makes sense, but is forbidden by the invisible pre-condition that `pre_low` is in range [0, 0xfffc]).